packages/store/src/utils/store-validators.ts

```diff
@@ -47,4 +48,16 @@ export abstract class StoreValidators {
 
     return meta;
   }
+
+  public static checkExtensibleBeforeDecorate(target: object, name: NGXS_DECORATOR): void {
+    if (!Object.isExtensible(target)) {
+      throw new Error(MESSAGES[CODE.CHECK_EXTENSIBLE_BEFORE_DECORATE](name));
+    }
+  }
+
+  public static checkIsFrozenBeforeDecorate(target: object, name: NGXS_DECORATOR): void {
+    if (Object.isFrozen(target)) {
```

Comment:
Isn't this check redundant? I mean who does `freeze` the class? I've never seen this:
```ts
@Component({ ... })
class AppComponent {
  @Select() counter$;

  constructor() {
    Object.freeze(this);
  }
}
```

Reply:
@markwhitfeld what do you think?

Reply:
Sorry, I also don't think that these checks add anything useful.

Reply:
done